### PR TITLE
Create `HttpFoundation\Request->toObject()` Function

### LINKED ISSUE #46864: Create `HttpFoundation\Request->toObject()` Function
### Description

Create a function, similar to `toArray()`, but instead to return an ***Array***, return an ***Object***

### Example

When the client send an **Json** to the Request, we manipulate the values, right? Image an scenario where we have to **Tranform** these values into an **Object:**
```php
$DataArray = $request->toArray();
$DataObject = (object) $DataArray;

$DataObject ->name;
$DataObject ->website;
```

So, the Feature Request is to create an `toObject` function, where will return an **Object**.
With the new function, we reduce the code:
```php
$Data = $request->toObject();

$Data->name;
$Data->website;
```

### PR DESCRIPTION
**Pull request by TheDevick**
| Q             | A
| ------------- | ---
| Branch?       | 6.2 
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Feat #46864
| License       | MIT
| Doc PR        | symfony/symfony-docs#16941

### CODE REVIEW

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -1503,6 +1503,22 @@ public function toArray(): array
...
+    /**
+     * Gets the request body decoded as array, typically from a JSON payload and transform to an Object.
+     *
+     * @throws JsonException When the body cannot be decoded to an object
+     */
+    public function toObject(): object
+    {
+        try {
+            $data = $this->toArray();
+        } catch (JsonException $message) {
```
Comment (FabienPapet) [inline src/Symfony/Component/HttpFoundation/Request.php:1515]: Not sure if this try catch is useful

**Review by FabienPapet — COMMENTED** [review]
I think `return (object) $this->toArray()`  is the way to go as the try catch is not useful

**Comment by TheDevick** [comment]
> I think `return (object) $this->toArray()` is the way to go as the try catch is not useful

So you mean remove the `try` and `catch` from the code?

**Comment by FabienPapet** [comment]
Yes, I think that you don't need to catch the exception from the toArray()

**Review by ro0NL — COMMENTED** [review]
i'd prefer casting in userland (as always :))

**Comment by fabpot** [comment]
I agree that there is no need for this method to be part of core as the use case is very specific and not broad enough. Thank you for proposing.

**Pull request closed without merging**